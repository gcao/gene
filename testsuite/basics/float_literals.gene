#!/usr/bin/env gene run

# Test: Float literal parsing and operations

# Basic float literals
(println 3.14)
(println 0.5)
(println 42.0)

# Float arithmetic
(println (+ 1.5 2.5))
(println (- 10.0 3.7))
(println (* 2.5 4.0))
(println (/ 10.0 3.0))

# Float comparisons - commented out (causes segfault)
# (println (.< 3.14 4.0))

# Mixed int/float operations
(println (+ 1 2.5))
(println (* 3 1.5))