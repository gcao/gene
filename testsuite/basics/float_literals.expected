3.14
0.5
42.0
4.0
6.3
10.0
3.333333333333333
3.5
4.5